buildkite/builds.go

```diff
@@ -84,6 +85,20 @@ type Build struct {
 	PullRequest *PullRequest `json:"pull_request,omitempty" yaml:"pull_request,omitempty"`
 }
 
+type MetaDataFilters struct {
+	MetaData map[string]string
+}
+
+// Encodes MetaData in the format expected by the buildkite API
+// Example: ?meta_data[some-key]=some-value
+func (mo MetaDataFilters) EncodeValues(parent_key string, v *url.Values) error {
+	for key, val := range mo.MetaData {
+		keyString := fmt.Sprintf("%s[%s]", parent_key, key)
+		v.Add(keyString, val)
```

Comment:
In here v is being added to as a url.Values list, which will perform URL "friendly" encoding when parsed. The value of this will need to be converted to a string in both the Get and List functions in order to force it to not use the URL encoded values for square brackets.